lib/net.js

```diff
@@ -825,11 +825,20 @@ protoGetter('bytesWritten', function bytesWritten() {
       bytes += Buffer.byteLength(el.chunk, el.encoding);
   });
 
-  if (data) {
-    if (data instanceof Buffer)
-      bytes += data.length;
-    else
-      bytes += Buffer.byteLength(data, encoding);
+  if (Array.isArray(data)) {
+    // was a writev, iterate over chunks to get total length
+    for (var i = 0; i < data.length; i++) {
+      const chunk = data[i];
+
+      if (chunk instanceof Buffer)
+        bytes += chunk.length;
+      else
+        bytes += Buffer.byteLength(chunk.chunk, chunk.encoding);
+    }
+  } else if (data instanceof Buffer) {
```

Comment:
I wonder if we could get away with using `data != undefined && typeof data !== 'string'` instead? If so, we could first branch on the first condition and then do a `if (typeof data !== 'string') { ... } else { ... }` to avoid duplicate `if (data)`-style checks.